### PR TITLE
Add variants of input fields

### PR DESCRIPTION
**Pull request by jefiedler**

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
+      <BasicTextInput idName="password" title="Passwort" inputType="password" />
+      <br />
+      <SalutationSelectorInput />
+      <br />
+      <DatePickerInput />
+      <br />
+      <TimePickerInput />
     </div>
   );
 }
```
Comment (phlgr) [inline client/src/App.js:22]: Example Code doesn't belong on the main site. Use Storybook for this type of example

File: client/src/stories/Component/Accordion/ParticipantAccordion.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import React from "react";
```
Comment (phlgr) [inline client/src/stories/Component/Accordion/ParticipantAccordion.js:1]: Why is this here and where do I find it? Maybe put a link in the description

File: client/src/stories/Component/Inputs/BasicTextInput.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        type={inputType}
+        onChange={onChange}
+      />
+    </div>
+  );
+}
+
+export default BasicTextInput;
+
+//Proptyps
```
Comment (phlgr) [inline client/src/stories/Component/Inputs/BasicTextInput.js:24]:
```suggestion
//PropTypes
```

File: client/src/stories/Component/Inputs/ParticipantInput.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import PropTypes from "prop-types";
+import BasicTextInput from "./BasicTextInput";
+import SalutationSelectorInput from "./SalutationSelectorInput";
+import DatePickerInput from "./DatePickerInput";
+import styled from "@emotion/styled";
```
Comment (phlgr) [inline client/src/stories/Component/Inputs/ParticipantInput.js:6]: Try not to mix different styling systems

File: client/src/stories/Component/Inputs/ParticipantInput.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        inputType="text"
+        onChange=""
+      />
+    </FormContainer>
+  );
+}
+
+export default ParticipantInput;
+
+//Proptyps
```
Comment (phlgr) [inline client/src/stories/Component/Inputs/ParticipantInput.js:62]:
```suggestion
//PropTypes
```

File: client/src/stories/Component/Inputs/ParticipantInput.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+  grid-template-rows: 1fr 1fr;
+`;
+
+const PersonalDataContainer = styled.div``;
+
+const AdressDataContainer = styled.div``;
+
+const MailDataContainer = styled.div``;
```
Comment (phlgr) [inline client/src/stories/Component/Inputs/ParticipantInput.js:77]: Why are these styled components?

File: client/src/stories/Component/Inputs/TimePickerInput.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+          }}
+        />
+      </MuiPickersUtilsProvider>
+    </div>
+  );
+}
+
+export default TimePickerInput;
+
+//Proptyps
```
Comment (phlgr) [inline client/src/stories/Component/Inputs/TimePickerInput.js:41]:
```suggestion
//PropTypes
```

**Review by phlgr — CHANGES_REQUESTED** [review]
Could you explain your used folder structure. Why are components in src/stories/...?
The stories look good to me.
🚀

File: client/package.json [hunk 8]
```diff
@@ -18,6 +20,7 @@
     "@testing-library/react": "^9.5.0",
     "@testing-library/user-event": "^7.2.1",
     "babel-loader": "^8.1.0",
+    "date-fns": "^2.16.1",
```
Comment (lmachens) [inline client/package.json:23]: I guess you don't need this. There is no import.

File: client/src/App.js [hunk 9]
```diff
@@ -1,9 +1,22 @@
...
+import SalutationSelectorInput from "../src/stories/Component/Inputs/SalutationSelectorInput";
+import TimePickerInput from "../src/stories/Component/Inputs/TimePickerInput";
 
 function App() {
   return (
     <div className="App">
-      <h1>Clean up for new styling</h1>
+      <h1>Verschiedene Inputfelder:</h1>
+      <BasicTextInput idName="firstName" title="Nachname" inputType="text" />
+      <br />
```
Comment (lmachens) [inline client/src/App.js:12]: `<br />`  is not for styling :)

File: client/src/stories/Component/Accordion/ParticipantAccordion.js [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  heading: {
+    fontSize: theme.typography.pxToRem(15),
+  },
+  details: {
+    width: "1350px",
+    height: "150px",
+    background: "#F2F2F2",
+    borderRadius: "5px",
+  },
+}));
```
Comment (lmachens) [inline client/src/stories/Component/Accordion/ParticipantAccordion.js:73]: To use emotion (or any other styled components like https://material-ui.com/styles/basics/#styled-components-api), you can try this:

```suggestion
const Heading = styled(Typography)`
  font-size: 0.9rem;
`;

const Details = styled(Typography)`
    width: 1350px;
    height: 150px;
    background: #F2F2F2;
    borderRadius: 5px;
`;
```

File: client/src/stories/Component/Inputs/BasicTextInput.js [hunk 11]
```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import PropTypes from "prop-types";
+import TextField from "@material-ui/core/TextField";
+import { makeStyles } from "@material-ui/core";
+
+function BasicTextInput({ idName, title, inputType, onChange }) {
+  const classes = useStyles();
+
+  return (
+    <div className={classes.root}>
```
Comment (lmachens) [inline client/src/stories/Component/Inputs/BasicTextInput.js:10]: I think you don't need this `div`.
Instead, style your `TextField` directly.

File: client/src/stories/Component/Inputs/DatePickerInput.js [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+
+  const handleDateChange = (date) => {
+    setSelectedDate(date);
+  };
+
+  const classes = useStyles();
+
+  return (
+    <div className={classes.root}>
+      <MuiPickersUtilsProvider utils={DateFnsUtils}>
```
Comment (lmachens) [inline client/src/stories/Component/Inputs/DatePickerInput.js:21]: Providers like this could be moved to your `App.js`, because you only need one in your application. If this component is rendered multiple times, it would create multiple providers.

**Review by lmachens — COMMENTED** [review]
gj
